docs/faq/code-analysis/how-long-does-it-take-for-my-repository-to-be-analyzed.md

```diff
@@ -1,7 +1,23 @@
 # How long does it take for my repository to be analyzed?
 
-Codacy usually takes under 5 minutes to analyze your repository, however it may take longer as this depends on the size of your repository.
+Codacy usually takes under 5 minutes to analyze your repository. It may however take longer, depending on a number of factors:
 
-Codacy relies on post-commit hooks sent by your Git provider to trigger the analysis after each push to the repository, so if your analysis is taking a lot of time to start [check that the Post-Commit Hook integration for your repository is enabled](../../repositories-configure/integrations/post-commit-hooks.md).
+-   **Whether it's the initial analysis of your repository**
```

Comment:
For this item, we can explain that the initial analysis takes longer because we're analyzing all files in the repository - while in subsequent commits we only analyze the files changed in the deltas. The exact details of this should be checked with someone from Engineering - @pedrocodacy would be a good candidate to quickly validate.

Reply:
Here's a suggestion:

```suggestion
-   **Whether it's the initial analysis of your repository**

    The initial analysis examines all files in the repository, while each subsequent commit triggers an analysis only on files changed in that commit.
```

Reply:
Yup this is what happens. Basically you can look at this as: on each analysis we will analyze every file that we havent analyzed yet in its current state, so:

- On the first analysis this means analyzing the entire repo (we wont have anything previously analyzed)
- On any other analysis we will look at the files and their content and try to match this with already existing results, if those results are missing for any subset of files we will analyze those files 
- Another note on this is: if configuration of tools change between analysis, the existing results for non-changed files that would be analysed for that tool will be discarded, and hence we will analyse these again using the new configuration (even if the file contents did not change)

Reply:
Thanks for confirming, @pedrocodacy! I'm also updating the suggestion above to include your third point.